frontend/apps/reader/modules/readerstatus.lua

```diff
@@ -258,6 +257,8 @@ function ReaderStatus:onMarkBook(mark_read)
     else
         self.settings.data.summary = {status = "complete"}
     end
+    self.settings:saveSetting("summary", self.settings.data.summary)
```

Comment:
That's a non obvious case, so good to add a comment (or someone may one day think this is not needed and could be removed):
`-- If History is called over Reader, it will read the file to get the book status, so save and flush`